VoyageAI improvements

fzowl opened the pull request:
Adding VoyageAI voyage-3.5 model
Adding VoyageAI voyage-context-3 model

fzowl committed:
Adding Voyage 3.5 models

fzowl committed:
Adding VoyageAI's voyage-context-3 model

dnwpark commented on edb/lib/ext/ai.edgeql:
```diff
@@ -562,6 +583,167 @@ CREATE EXTENSION PACKAGE ai VERSION '1.0' {
+
+    create abstract type ext::ai::Voyage3LiteEmbedModel
+        extending ext::ai::EmbeddingModel
+    {
```
Another old model

dnwpark commented on edb/lib/ext/ai.edgeql:
```diff
@@ -562,6 +583,167 @@ CREATE EXTENSION PACKAGE ai VERSION '1.0' {
+        alter annotation
+            ext::ai::embedding_model_max_batch_tokens := "320000";
+        alter annotation
+            ext::ai::embedding_model_max_output_dimensions := "1024";
```
Should be 2048?

dnwpark reviewed (commented):
Thanks for the contribution! Here are some notes from my testing too.

The max vector size is currently 2000, this causes the indexing to get truncated.

Also it seems that the default embedding length for a lot of models is 1024. Some kind of parameter seems to be missing from the request, and this causes an error when searching text. It's probably fine to just use 1024 for max size.

fzowl committed:
Corrections due to comments.

fzowl committed:
Corrections due to comments.
Documentation update

fzowl commented:
@dnwpark Sorry, i missed your review and comments. Can you please check the changes now?

fzowl committed:
Adding VoyageAi v4 models

fzowl committed:
Address review feedback from dnwpark

- Add deprecated annotation to legacy VoyageAI models
- Fix VoyageContext3 max output dimensions to 2048
